src/transformers/modeling_utils.py

```diff
@@ -1580,6 +1584,32 @@ def _autoset_attn_implementation(
             # If a config is passed with a preset attn_implementation, we skip the automatic dispatch and use the user-provided config, with hard checks that the requested attention implementation is available.
             requested_attn_implementation = config._attn_implementation_internal
 
+        # Composite models consisting of several PretrainedModels have to specify attention impl as a dict
+        # where keys are sub-config names. But most people will specify one `str` which means that should dispatch it
+        # for all sub-models.
```

Comment:
It makes sense, sometimes you want output hidden states of one module but not the other one, so I guess it's minimul flexibility.

Reply:
but IMO we should propagate here